### PR TITLE
Remove __IMAGE_SUPPORT__ macro for SPIR

### PR DESCRIPTION
**Pull request by haonanya**
Signed-off-by: haonanya <haonan.yang@intel.com>

### CODE REVIEW

File: patches/clang/0004-Remove-__IMAGE_SUPPORT__-macro-for-SPIR.patch [hunk 1]
```diff
@@ -0,0 +1,42 @@
+From e505c407951ec79a04e76a2592a9e3528180f0c4 Mon Sep 17 00:00:00 2001
+From: haonanya <haonan.yang@intel.com>
+Date: Sat, 8 May 2021 11:18:46 +0800
+Subject: [PATCH] Remove __IMAGE_SUPPORT__ macro for SPIR
```
[the remark this answers is not in the document]
Comment (haonanya) [reply patches/clang/0004-Remove-__IMAGE_SUPPORT__-macro-for-SPIR.patch]: done.

[the remark this answers is not in the document]
Comment (hewj03) [reply patches/clang/0004-Remove-__IMAGE_SUPPORT__-macro-for-SPIR.patch]: I'm very sorry I withdraw my suggestion. If the fix is right, then the fix is intended for all devices.

**Comment by haonanya** [comment]
Kindly ping @hewj03 @cdai2 @fzou1.  Also same patches for [branch 120](https://github.com/intel/opencl-clang/pull/242), [branch 110](https://github.com/intel/opencl-clang/pull/245), [branch 100](https://github.com/intel/opencl-clang/pull/244), [branch master](https://github.com/intel/opencl-clang/pull/246)
Thanks very much.

**Review by hewj03 — APPROVED** [review]
LGTM

**Pull request merged**